semconv/internal/v2/http_test.go

```diff
@@ -76,7 +88,6 @@ func TestHTTPSClientRequest(t *testing.T) {
 		t,
 		[]attribute.KeyValue{
 			attribute.String("http.method", "GET"),
-			attribute.String("http.flavor", "1.0"),
```

Comment:
Why is this attribute removed from request spans?

Reply:
Sorry, I just misunderstood that I couldn't overwrite the attribute when fill in the attribute in ClientResponse.

Reply:
@dmathieu 

No, I may have misunderstood the intent of your question.

I interpreted your question as "Why do you change the behavior of ClientRequest? And when I checked the behavior again at hand, even though I filled in the http.flavor attribute twice (`ClientRequest` and `ClientResponse`), the one I put in later (`http2.0` filled in `ClientResponse`) took precedence, so I thought I had made a behavior check error when I sent the first change. And I pushed the fix commit.

However, this was just a mistake in the build when I checked the behavior this time, and it appears that `http1.1` is still selected as the final span trace when `http.flavor` is filled in `ClientRequest`, just as it was in the first change.

I need to understand the specification, but apart from that, may I ask again what you meant by your question.

Reply:
> However, this was just a mistake in the build when I checked the behavior this time, and it appears that http1.1 is still selected as the final span trace when http.flavor is filled in ClientRequest, just as it was in the first change.

My apologies again for the repetition. This was wrong.
I think I should revert the change of `ClientRequest` behavior.

> I interpreted your question as "Why do you change the behavior of ClientRequest?"
> And when I checked the behavior again at hand, even though I filled in the http.flavor attribute twice (ClientRequest and ClientResponse), the one I put in later (http2.0 filled in ClientResponse) took precedence, so I thought I had made a behavior check error when I sent the first change.

So, if this interpretation is correct, I would be grateful to know.

When I run [otel-go-contrib example](https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/instrumentation/net/http/otelhttp/example) with the revert of `ClientRequest` change, I got this.
```
$ go run client/client.go
Sending request...
# the span struct I got
span: &{{0 0} {[198 4 41 43 97 207 33 169 73 71 239 188 33 112 138 164] [167 246 235 174 189 96 91 123] 1 {[]} false} 3 HTTP GET {13904739827806336174 1977246 0x94c280} {0 0 <nil>} {0 } 0 {[198 4 41 43 97 207 33 169 73 71 239 188 33 112 138 164] [169 242 129 85 37 196 162 38] 1 {[]} false} [{http.method {4 0 GET <nil>}} {http.flavor {4 0 1.1 <nil>}} {http.url {4 0 https://localhost:7777/hello <nil>}} {net.peer.name {4 0 localhost <nil>}} {net.peer.port {2 7777  <nil>}} {http.flavor {4 0 2.0 <nil>}} {http.status_code {2 200  <nil>}} {http.response_content_length {2 14  <nil>}}] 0 {[] 128 0} {[] 128 0} <nil> 0xc0000e0940}
Response Received: Hello, world!



Waiting for few seconds to export spans ...

{
        "Name": "HTTP GET",
        "SpanContext": {
                "TraceID": "c604292b61cf21a94947efbc21708aa4",
                "SpanID": "a9f2815525c4a226",
                "TraceFlags": "01",
                "TraceState": "",
                "Remote": false
        },
        "Parent": {
                "TraceID": "c604292b61cf21a94947efbc21708aa4",
                "SpanID": "a7f6ebaebd605b7b",
                "TraceFlags": "01",
                "TraceState": "",
                "Remote": false
        },
        "SpanKind": 3,
        "StartTime": "2023-02-28T18:11:43.849310894+09:00",
        "EndTime": "2023-02-28T18:11:43.866316235+09:00",
        "Attributes": [
                {
                        "Key": "http.method",
                        "Value": {
                                "Type": "STRING",
                                "Value": "GET"
                        }
                },
                {
                        "Key": "http.flavor",
                        "Value": {
                                "Type": "STRING",
                                "Value": "2.0"
                        }
                },
                {
                        "Key": "http.url",
                        "Value": {
                                "Type": "STRING",
                                "Value": "https://localhost:7777/hello"
                        }
                },
                {
                        "Key": "net.peer.name",
                        "Value": {
                                "Type": "STRING",
                                "Value": "localhost"
                        }
                },
                {
                        "Key": "net.peer.port",
                        "Value": {
                                "Type": "INT64",
                                "Value": 7777
                        }
                },
                {
                        "Key": "http.status_code",
                        "Value": {
                                "Type": "INT64",
                                "Value": 200
                        }
                },
                {
                        "Key": "http.response_content_length",
                        "Value": {
                                "Type": "INT64",
                                "Value": 14
                        }
                }
        ],
        "Events": null,
        "Links": null,
        "Status": {
                "Code": "Unset",
                "Description": ""
        },
        "DroppedAttributes": 0,
        "DroppedEvents": 0,
        "DroppedLinks": 0,
        "ChildSpanCount": 0,
        "Resource": [
                {
                        "Key": "service.name",
                        "Value": {
                                "Type": "STRING",
                                "Value": "unknown_service:client"
                        }
                },
                {
                        "Key": "telemetry.sdk.language",
                        "Value": {
                                "Type": "STRING",
                                "Value": "go"
                        }
                },
                {
                        "Key": "telemetry.sdk.name",
                        "Value": {
                                "Type": "STRING",
                                "Value": "opentelemetry"
                        }
                },
                {
                        "Key": "telemetry.sdk.version",
                        "Value": {
                                "Type": "STRING",
                                "Value": "1.13.0"
                        }
                }
        ],
        "InstrumentationLibrary": {
                "Name": "go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp",
                "Version": "semver:0.39.0",
                "SchemaURL": ""
        }
}
{
        "Name": "say hello",
        "SpanContext": {
                "TraceID": "c604292b61cf21a94947efbc21708aa4",
                "SpanID": "a7f6ebaebd605b7b",
                "TraceFlags": "01",
                "TraceState": "",
                "Remote": false
        },
        "Parent": {
                "TraceID": "00000000000000000000000000000000",
                "SpanID": "0000000000000000",
                "TraceFlags": "00",
                "TraceState": "",
                "Remote": false
        },
        "SpanKind": 1,
        "StartTime": "2023-02-28T18:11:43.849282044+09:00",
        "EndTime": "2023-02-28T18:11:43.866325383+09:00",
        "Attributes": [
                {
                        "Key": "peer.service",
                        "Value": {
                                "Type": "STRING",
                                "Value": "ExampleService"
                        }
                }
        ],
        "Events": null,
        "Links": null,
        "Status": {
                "Code": "Unset",
                "Description": ""
        },
        "DroppedAttributes": 0,
        "DroppedEvents": 0,
        "DroppedLinks": 0,
        "ChildSpanCount": 1,
        "Resource": [
                {
                        "Key": "service.name",
                        "Value": {
                                "Type": "STRING",
                                "Value": "unknown_service:client"
                        }
                },
                {
                        "Key": "telemetry.sdk.language",
                        "Value": {
                                "Type": "STRING",
                                "Value": "go"
                        }
                },
                {
                        "Key": "telemetry.sdk.name",
                        "Value": {
                                "Type": "STRING",
                                "Value": "opentelemetry"
                        }
                },
                {
                        "Key": "telemetry.sdk.version",
                        "Value": {
                                "Type": "STRING",
                                "Value": "1.13.0"
                        }
                }
        ],
        "InstrumentationLibrary": {
                "Name": "example/client",
                "Version": "",
                "SchemaURL": ""
        }
}
```

Reply:
As specification, filling a key with the same name multiple times is not a problem in itself, but the expectation is that you will win later.

> Setting an attribute with the same key as an existing attribute SHOULD overwrite the existing attribute's value.

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#set-attributes

Reply:
reverted the ClientRequest's behavior change
https://github.com/open-telemetry/opentelemetry-go/pull/3778/commits/67e2447a67d804436f374235751673ed8de12b0e